semantic-analysis/src/Analysis/Data/Snoc.hs

```diff
@@ -0,0 +1,7 @@
+{-# LANGUAGE DeriveTraversable #-}
+module Analysis.Data.Snoc
```

Comment:
I personally would just put this in `Data.Snoc`, or maybe `Data.List.Snoc`, since there’s nothing inherently analysis-related about it.

Reply:
The reorg of the project involved putting all modules inside package-scoped hierarchies. I'm pretty happy with this, because:

1. collisions are extremely unlikely, and, more importantly,
2. it's crystal clear where a given module came from, including whether it's ours or someone else's.